### PR TITLE
added install-all.sh

### PR DESCRIPTION
**Pull request by laluka**
Hi folks !
Thanks for the awesome work on findings these gems ! 

So, this is just a draft, and might not be the best way to do it, but it could be used as a base for tools installation ! :)

A few ideas of improvement: 
- Some tools are conflicting, maybe, like for history : fzf XOR mcfly
- The install procedure relies too much on dependencies, maybe for every tool parse the github release page and install the package ?
- Have some sort of arg parsing to select the tools to install

That being said.. Have a nice day!
Cheers, lalu

### CODE REVIEW

**Comment by slavaGanzin** [comment]
```
sudo apt install -y fzf
```

So it's not `install-all.sh` it's `install-all-on-ubuntu.sh`.

**Comment by laluka** [comment]
> ```
> sudo apt install -y fzf
> ```
> 
> So it's not `install-all.sh` it's `install-all-on-ubuntu.sh`.

There are so many install methods for these tools, it's a first start ! :)

Maybe with the approach of automated parsing of every release page, we can at least cover unix/macos

**Comment by slavaGanzin** [comment]
> Maybe with the approach of automated parsing of every release page, we can at least cover unix/macos

[AUR](https://wiki.archlinux.org/title/Arch_User_Repository) is user supported recipes for installation for Arch. A couple of times I used them to guide my installation on other *nix systems. They are easily parsable, standardized and half-time isn't Arch speciefic, so they are easier to parse then release pages.

BTW, you can install fzf with just git

**Comment by SuperSandro2000** [comment]
> There are so many install methods for these tools, it's a first start ! :)

Some of them are in Debian, some of them are in arch, some require go or rust and probably most of them are in nixpkgs. There are to many variants to cover here. I think this is a maintenance rabbit hole and not worth the effort.

File: install-all.sh [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+if ! command -v cargo &> /dev/null
+then
+    echo "cargo (rust) could not be found, install at https://doc.rust-lang.org/cargo/getting-started/installation.html or run"
+    echo "sudo apt install -y cargo"
+    echo "Also, ensure ~/.cargo/bin/ is in your PATH variable"
+    exit
+fi
+
+if ! command -v docker &> /dev/null
```
Comment (georgettica) [inline install-all.sh:27]: With the issues docker is exhibiting now, maybe also have alternatives to docker?

File: install-all.sh [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+else
+    sudo snap install procs
+fi
+
+# cheat
+if command -v cheat &> /dev/null
+then
+    echo "cheat already installed"
+else
+    go get -u github.com/cheat/cheat/cmd/cheat
```
Comment (georgettica) [inline install-all.sh:77]: In newer versions of go you can use
``` go install ```

Please investigate and bote your results

File: install-all.sh [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+fi
+
+# broot
+if command -v br &> /dev/null
+then
+    echo "br already installed"
+else
+    wget https://dystroy.org/broot/download/x86_64-linux/broot
+    chmod +x broot
```
Comment (georgettica) [inline install-all.sh:86]: Use one tool if possible

You used curl before and now you are using wget

File: install-all.sh [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+# delta
+if command -v delta &> /dev/null
+then
+    echo "delta already installed"
+else
+    cargo install git-delta
+fi
+
+# dog
+if grep -F "alias dog" ~/.zshrc &> /dev/null
```
Comment (georgettica) [inline install-all.sh:99]: If you are checking only in zshrc, why are you adding to the bashrc aswell?

File: install-all.sh [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+else
+    sudo apt install -y httpie
+fi
+
+# glances
+if grep -F "alias glances" ~/.zshrc &> /dev/null
+then
+    echo "glances already installed"
+else
+    echo 'alias glances="docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:ro --pid host --network host -it nicolargo/glances:latest-full"' >> ~/.zshrc
```
Comment (georgettica) [inline install-all.sh:120]: I am not sure mounting the docker socket is a security best-practice

File: install-all.sh [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    sudo apt install -y fzf
+fi
+
+# gping
+if command -v gping &> /dev/null
+then
+    echo "gping already installed"
+else
+    echo "deb http://packages.azlux.fr/debian/ buster main" | sudo tee /etc/apt/sources.list.d/azlux.list
+    wget -qO - https://azlux.fr/repo.gpg.key | sudo apt-key add -
```
Comment (georgettica) [inline install-all.sh:171]: Same wget/curl here

File: install-all.sh [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+fi
+
+if ! command -v cargo &> /dev/null
+then
+    echo "cargo (rust) could not be found, install at https://doc.rust-lang.org/cargo/getting-started/installation.html or run"
+    echo "sudo apt install -y cargo"
+    echo "Also, ensure ~/.cargo/bin/ is in your PATH variable"
+    exit
+fi
+
```
Comment (georgettica) [inline install-all.sh:26]: Verify dependencies for the tools (curl) exist before running the script

File: install-all.sh [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+    wget "$LATEST"
+    sudo apt install -y ./curlie*.deb
+fi
+
+# gtop
+if grep -F "alias gtop" ~/.zshrc &> /dev/null
+then
+    echo "gtop already installed"
+else
+    echo 'alias gtop="docker run --rm -it --name gtop --net=host --pid=host aksakalli/gtop"' >> ~/.zshrc
```
Comment (georgettica) [inline install-all.sh:191]: Why is the running with the hosts pid?

File: install-all.sh [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+else
+    curl -sS https://webinstall.dev/zoxide | bash
+fi
+
+# bottom
+if command -v btm &> /dev/null
+then
+    echo "btm already installed"
+else
+    LATEST=$(curl -s https://github.com/ClementTsang/bottom/releases | grep -oP "/[a-zA-Z0-9.:_/]+_amd64.deb" | grep -viE "nightly" | head -n 1 | sed "s#^#https://github.com&#g")
```
Comment (georgettica) [inline install-all.sh:208]: I am not sure this is the best way of pulling packages from GitHub

**Review by georgettica — CHANGES_REQUESTED** [review]
I made a good effort in checking this change

It is highly opinionated and these opinions should be communicated before any action in the script takes place

Anyhow I hope my review is helpful for you

File: install-all.sh (hunk 1 above)
Comment (slavaGanzin) [reply install-all.sh:27]: @georgettica offtopic: what are the issues?

Comment (georgettica) [reply install-all.sh:27]: Will discuss on a private channel (via LinkedIn)

**Comment by ibraheemdev** [comment]
Don't really want to encourage installing every single tool on this list, sorry.

**Comment by georgettica** [comment]
just for the sake of argument, how would you want to separate them? is there a graduation process from "not on the list" to "on the list"

**Comment by mgrzaslewicz** [comment]
@laluka for ubuntu provision and updates of provisioned stack you can check
https://github.com/mgrzaslewicz/ubuntu-workstation-provision
With an easy way to include/exclude things.

I created it to have repeatable ubuntu environment with easy to maintain approach based on ansible.
Many of the tools included were inspired by modern-unix repo.

And for macos there is https://github.com/mgrzaslewicz/macos-workstation-provision

**Comment by laluka** [comment]
Thanks @mgrzaslewicz ;)
But this is 3 years old, in the meantime I made my 'own offensive distro' in the form of a bunch of playbooks as well! ;)

If you want to have a look : https://github.com/laluka/lalubuntu

Cheers, and thanks for the tips still! 🌹🤙

**Pull request closed without merging**